### ISSUE TITLE
Locally cache emote images for faster load times and reduced load on CDN servers

### ISSUE THREAD

**Issue by Xzensi** [Xzensi/NipahTV#171]

**Comment by cmd430** [comment]
I don't think you need to do anything for/about this, browser will only request the emote once for the session then use cache for additional requests of the same emote image as shown here ![image](https://github.com/user-attachments/assets/8214d19e-f12c-4e26-888a-f0e672052df9)

**Comment by Xzensi** [comment]
> I don't think you need to do anything for/about this, browser will only request the emote once for the session then use cache for additional requests of the same emote image as shown here ![image](https://private-user-images.githubusercontent.com/2668906/371980521-8214d19e-f12c-4e26-888a-f0e672052df9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2OTA3NTcsIm5iZiI6MTcyNzY5MDQ1NywicGF0aCI6Ii8yNjY4OTA2LzM3MTk4MDUyMS04MjE0ZDE5ZS1mMTJjLTRlMjYtODg4YS1mMGU2NzIwNTJkZjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MzBUMTAwMDU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzQ5NDkwYzg1ZTExZGE5NjY2MGRkZTc2MmQ1YjEyNDk5YmEzNGFiOWFhYzkxYzYwMzU1ZDBkMjFjOWUyZjU4ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.g1OtG4O-ylRYSj-ppnh8vwE3fUPZss-yVacha5F_HDs)
> 
> ETA: I don't personally use Nipah (I have my own private tweak) but this should be true for Nipah unless you are doing something to invalidate the URLs for emotes (by adding query params for example)

That's certainly true, however on every browser reload it'll have discarded this cache. I can make emote loading nearly instant and at same time further reduce load on CDN's by caching it in CacheStorage as well. The reason for the particularily high P2 priority is because I noticed quite a few 7TV CDN degraded service issues with people reporting that 7TV emotes weren't loading, or loading uncomfortably slow. This would at least make it more reliable and stable for when 7TV does have degraded service issues due to high load.

But thinking of lowering priority and focus on other things instead.
